Big Red Last Stand Update

PalJohnson opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->

## About The Pull Request

- This update is seeking to shake up the survivor gameplay by providing unique scenarios for survivors. This update adds several survivor holdouts (neutral & hostile) for survivors. In addition, I added a minor bugfix for synth survivor,

This update got delayed due to gitkraken nonsense, the bot closing my mr, and etc. Sorry for delay

## Why It's Good For The Game

- One of the most underrated aspects of survivors is the ability to be able to do a last stand. The current meta shift revolves around LZ holds or roaming if that's not possible. This update should encourage survivors not to leave their place of work and instead fight and in most cases die for their company if they feel inclined to as opposed to roaming or cheesing out LZ holds.

## Non-Goals

- Giving survivors additional gear. Given that their gear is random and spread across the maps survivors will now have a focal point to restock on valued items in a localized area and runoff. Although survivors are expected to stay and fight to the last colonist, nothing stops them from looting the place. In the end of the day its player agency and I don't think it would be good to force them to do something they don't want to do like die in the bar, library or marshalls. Given that most the CLF insert have 25% spawn rate, as well as the dnd, insert this shouldn't be a problem. Given the frequency of the bar holdout, I have purposely given them fewer resources.
- 
- Xenomorphs have a harder time killing survivors. Although killing or capturing survivors is not a xenomorph goal as they can ignore them entirely. Having a holdout at places like bar or even libraries for that matter could prove to be a nuisance due to premade defenses and gear being available. As stated previously nothing guarantees the survivors staying there in the first place so they could be unmanned and easily melted. Don't assault a survivor holdout alone.



## Changelog

_**- Maps**_

This update as mentioned adds 3 new survivor holdouts for survivors that will randomly spawn throughout the map. Currently, there are only 3 being added but more will be added in the future. The colony didn't get raptured it was a bloody mess with people crying and fighting a hell hole. The MR has been balanced using the CLF crashed ship insert on LV-624 as an example as to how much medical, guns and other items will be spawned in for survivors. These are prepared positions so they should have ok gear and protection if we actually want to encourage people to hold it.

_**CLF Marshalls Take Over**_ **(25% Chance)**
![image](https://user-images.githubusercontent.com/103712112/164294844-6ce804ae-a33d-4025-a5de-163e7f861357.png)

- As stated previously the CLF have invaded the colony and have taken over the last bastion of the authority of the colony that is the colonial marshall's offices and armory. After the successful takeover, the CLF decided to make the ruins of the office into their own area of operation for future CLF activities in the colony. much like the crashed ship insert, they have been equipped with identical gear to their LV-624 counterparts with minor exceptions. The exception is materials to repair and potentially expand their area of influence. The barricades for the most part are worn out and need some repairs if they expect to live long enough.


_**Dungeons and Dragons Last Stand**_ **(25% Chance)**
![image](https://user-images.githubusercontent.com/103712112/164294913-9de7de63-e4d8-4ee8-ae57-3072e87e1e8e.png)
- This nightmare insert was inspired by the event held by jarek where survivors attempted to have a sleepover in the LV nexus dormitories only to be slaughtered. This insert mirrors that in that colonists are hosting a dnd session in the local library. Given that the civilian survivor is rather lackluster I have gone ahead and supplied them with cosplay material so that they can arm themselves and carry out their dnd session in style. Much of it is fluff in many regards with the exception of the illegally smuggled merc gear. The balance of the insert is similar in the crashed ship insert but the barricades are repaired and the shutters closed.

_**The Bar Last Stand**_ **_(40% Chance)_**
![image](https://user-images.githubusercontent.com/103712112/164294934-098f1307-c186-4cd0-a4b5-39aaf8c98e85.png)
- This insert reflects the valiant efforts of the colonist to have a last stand within the colony bar. The bar (according to the colonist handbook) was designated as an emergency rally point for all personnel and staff in the event of an unforeseen event. They come equipped with standard wooden barricades with building material should they want to replace them. In addition, they have ammo, medical supplies, and liquid courage to help them in their endeavor.

_**Minor Updates**_

- Synth survivor now has colonist radio. Previously in the synth survivor update, I accidentally gave them the wrong radio. This prevented them from communicating with survivors as they had a unique radio. The auto lathes produce marine radios so they couldn't find an adequate communication device.

- Colonist survivor now has the colonist jumpsuit. They had PJs for some reason. Odd

🆑 TheFlyingFlail
add: Added three new survivor holdout nightmare inserts for big red
add: Added clf survivor to big red with their own unique story to go with it
expansion: Expanded existing mechanics or gameplay
fix: fixed synth survivors not having colonists' radio
fix: fixed civilian survivor not having actual colonist clothing
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

PalJohnson commented:
Reuploaded and fixed the mr. The bot made it stale on two occasions so I stopped working on it. I finally got back to it and fixed the bugs in it. Hopefully, this works.

c4xmaniac commented:
SPEED MERGE YOU LET IT DIE LAST TIME DEVS MERGE MERGE MERGE GO GO GO!!!!!

anclandor commented:
Add more dice!

DesertFox2 commented:
I think it would be cool to add a further chance of some of the wooden cades to be replaced by heavily damaged metal ones

stanalbatross commented:
![image](https://user-images.githubusercontent.com/66756236/177654911-73b17fa2-1b23-437b-99b9-4b4b27aa8940.png)
this makes little sense, how did they get all that stuff in there?

PalJohnson commented:
> 

The items here are based off the CLF nightmare insert in the LV-624 crashed ship insert. There is another surgical bed in medbay which is a little closer but it is rarely used even when marine take it over with relative ease at the beginning of the round as they prefer to preform surgery from within the fob. I have gone in and replaced the surgical bed with normal beds aswell as the scaner aswell.

![image](https://user-images.githubusercontent.com/103712112/177835903-008ef0ac-905a-4937-9712-4f65abeaba2f.png)

stanalbatross commented on code/game/objects/effects/landmarks/survivor_spawner.dm:
```diff
@@ -58,3 +58,17 @@
+	equipment = /datum/equipment_preset/survivor/civilian
+	intro_text = list("<h2>You are a civilian participating in a dungeons and dragons game!</h2>",\
+	"<span class='notice'>You are NOT aware of the xenomorph threat due to the rather engaging dnd game going on.</span>",\
+	"<span class='danger'>Your primary objective is to have a fun time and survive.</span>")
```
```suggestion
	"<span class='danger'>Your primary objective is to have a fun time and survive. If you want to assault the hive - adminhelp.</span>")
```

stanalbatross commented on code/game/objects/effects/landmarks/survivor_spawner.dm:
```diff
@@ -58,3 +58,17 @@
+	intro_text = list("<h2>You are a civilian participating in a dungeons and dragons game!</h2>",\
+	"<span class='notice'>You are NOT aware of the xenomorph threat due to the rather engaging dnd game going on.</span>",\
+	"<span class='danger'>Your primary objective is to have a fun time and survive.</span>")
+	story_text = "You are a goup of colonist who regularly meet up to play dnd in the library every friday night. You alongside your fellow colonist brought various costumes, food, drinks, and props (very real guns) to help get you all into character. You aren't very exceptional in your skills but you do have the one thing that bests even the mightiest demons. That special power imagination!"
```
```suggestion
	story_text = "You are a group of colonists who regularly met up to play DnD in the library every Friday night. You, alongside your fellow colonists brought various costumes, food, drinks, and props (very real guns) to help get you all into character. You aren't very exceptional in your skills but you do have the one thing that bests even the mightiest demons - that special power, imagination!"
```

PalJohnson replied:
applied suggestions

stanalbatross replied:
have you pushed? doesn't seem applied

stanalbatross commented on code/game/objects/effects/landmarks/survivor_spawner.dm:
```diff
@@ -58,3 +58,17 @@
+	intro_text = list("<h2>You are a survivor of a colonial uprising!</h2>",\
+	"<span class='notice'>You are NOT aware of the xenomorph threat.</span>",\
+	"<span class='danger'>Your primary objective is to heal up and survive. If you want to assault the hive - adminhelp.</span>")
+	story_text = "You are a soldier of Colonial Liberation Front. Your ship received a distress signal from a planet bordering the CLF controlled space under USCM control. Ready and willing to save poor colonists from parasitic tyrants, you and your team boarded small ship called Marie Curie. You take over the colonial marshalls office and send an all clear signal to the nearby PMC distress team how ...
```
```suggestion
	story_text = "You are a soldier of the Colonial Liberation Front. Your ship received a distress signal from a planet bordering CLF controlled space under USCM control. Ready and willing to save poor colonists from parasitic tyrants, you and your team boarded a small ship called the Marie Curie. You took over the Colonial Marshal's office and sent an all clear signal to the nearby PMC distress team, however an unknown entity has responded to the call. You dont know who they are but they will likely identify you as a foe."
```
why are they trying to call pmcs, anyway? clf and pmcs are not aligned.

stanalbatross reviewed (changes requested):
![image](https://user-images.githubusercontent.com/66756236/179203829-f21fc31a-6683-4d3a-b58f-2d8554a6d9f9.png)
just rods etc is a little boring, you can also leave some electronic debris, ashes, wires, metal sheets, airlock boards around these destroyed airlocks (not to mention destroyed airlocks don't drop rods, nor are they used in airlock construction)

stanalbatross reviewed (changes requested):
![image](https://user-images.githubusercontent.com/66756236/179204205-7978aa87-002d-4527-9451-39b311962576.png)
this is a LOT of meds + a nailgun which basically means they will never have to venture out of the bar to get other supplies

stanalbatross reviewed (changes requested):
![image](https://user-images.githubusercontent.com/66756236/179204347-48381148-0bfc-4554-a62e-f3d770111ce3.png)
same problem here

giving survs easy access to meds and the nailgun (super powerful) trivialises the "gathering" component of their gameloop and lets them just sit in their fort the entire time

stanalbatross commented:
the CLF ship insert on LV has a lot of loot because chance of survival is very low - there is almost no escape from it, so you are trapped in the ship, plus you're surrounded on all sides and you're right next to the hive. The supplies in the ship are all that you can possibly get, so it somewhat makes sense to give out a lot of them

in contrast, marshalls is far from any hive spots, can easily be escaped from, and is right next to the entire rest of the colony, I therefore don't think that a CLF marshalls insert needs as many automatically handed out guns and pieces of gear as the LV one

PalJohnson commented:
> 9\. Also drop some welding masks or goggles so survivors don't go blind if they do choose to stick around and repair.

![image](https://user-images.githubusercontent.com/103712112/179588838-3bcc50e7-1b8b-43de-a4c6-257d61b7054e.png)
![image](https://user-images.githubusercontent.com/103712112/179589257-e7005fa5-b1f1-4411-a60c-b07abe1a421a.png)
![image](https://user-images.githubusercontent.com/103712112/179589577-f652b66e-e751-4f33-a54c-76086c9e8139.png)
![image](https://user-images.githubusercontent.com/103712112/179590198-babf8e3e-851e-48c3-857a-038488e81625.png)

Applied changes as requested. As per dealing with people running away, I can only say we need to incentivize people to stay in the area with loot and a purpose. I don't think we should force players to stay there (like in the cabin spawn in lv). i removed the m60 for the mar40 lmg as it is featured on the mining encampment south of the map. As for the ammo box there exist no ammo box for the mar40 so I just doubled the amount of ammo. I replaced the security barricades with wooden barricades. Attachments have been scattered. For the security hold the walls have been reinforced to address concerns about being melted easily. I will say survivors have been able to hold the area in the past but with the advent of T2s being able to break walls and acid holdouts don't tend to last a long time without a significant amount of materials to plug holes. This was the middle ground.

PalJohnson commented:
https://github.com/cmss13-devs/cmss13/pull/706 (New PR. Timeline screwed up on this pr. New pr fixed issue. Funny merging magic)
